distributed/utils.py

```diff
@@ -128,6 +128,22 @@ def get_ipv6(host='2001:4860:4860::8888', port=80):
     return _get_ip(host, port, family=socket.AF_INET6, default='::1')
 
 
+import socket
+import fcntl
+import struct
+
+def get_ip_interface(ifname):
+    # http://stackoverflow.com/a/24196955/616616 by @martin-konecny
```

Comment:
Wow. How about using [psutil.net_if_addrs](https://pythonhosted.org/psutil/#psutil.net_if_addrs) instead?
Or, if that doesn't work for us, there's the dedicated [netifaces](https://pypi.python.org/pypi/netifaces) package.